src/mpid/ch4/src/ch4_progress.h

```diff
@@ -34,8 +34,9 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int flags)
     }
 #endif
 
-    if (OPA_load_int(&MPIDI_CH4_Global.registered_progress_hooks) && (flags & MPIDI_PROGRESS_HOOKS)) {
-        for (i = 0; i < MAX_PROGRESS_HOOKS; i++) {
+    if (num_of_hooks = OPA_load_int(&MPIDI_CH4_Global.registered_progress_hooks) &&
```

Comment:
Should be `==`.

Reply:
No, I think `=` is correct.

Reply:
Hmm, on second read maybe not.

Reply:
Although, putting the `num_of_hooks = OPA_load_int(...)` part as a different statement might be cleaner (and easier to parse).

Reply:
Agreed.

Reply:
The if condition will be on the value of the counter. We just want to load the current counter value into `num_of_hooks` and use that value. So `=` would be what we want?

Reply:
FWIW, this stackoverflow answer adds some nice context to whether or not this is a good idea to do assignment in conditional. https://stackoverflow.com/questions/17681535/variable-assignment-in-if-condition/18450261#18450261

Reply:
Seems like using the parentheses around the assignment would've been ok in this case. But I put it into a separate statement so it is cleaner. Just force-pushed the commit with the changes.